classes/controller/FrontController.php

```diff
@@ -1229,29 +1232,57 @@ public function addJqueryPlugin($name, $folder = null, $css = true)
      */
     protected function recoverCart()
     {
-        if (($id_cart = (int) Tools::getValue('recover_cart')) && Tools::getValue('token_cart') == md5(_COOKIE_KEY_ . 'recover_cart_' . $id_cart)) {
-            $cart = new Cart((int) $id_cart);
-            if (Validate::isLoadedObject($cart)) {
-                $customer = new Customer((int) $cart->id_customer);
-                if (Validate::isLoadedObject($customer)) {
-                    $customer->logged = true;
-                    $this->context->customer = $customer;
-                    $this->context->cookie->id_customer = (int) $customer->id;
-                    $this->context->cookie->customer_lastname = $customer->lastname;
-                    $this->context->cookie->customer_firstname = $customer->firstname;
-                    $this->context->cookie->logged = true;
-                    $this->context->cookie->check_cgv = 1;
-                    $this->context->cookie->is_guest = $customer->isGuest();
-                    $this->context->cookie->passwd = $customer->passwd;
-                    $this->context->cookie->email = $customer->email;
-                    $this->context->cookie->id_guest = (int) $cart->id_guest;
-
-                    return $id_cart;
-                }
-            }
+        if (!Tools::isSubmit('recover_cart')) {
+            return false;
         }
 
-        return false;
+        // Get ID cart from URL
+        $id_cart = (int) Tools::getValue('recover_cart');
+
+        // Check if token in URL matches, otherwise, ignore it, probably malicious intentions
+        if (Tools::getValue('token_cart') != md5(_COOKIE_KEY_ . 'recover_cart_' . $id_cart)) {
```

Comment:
```suggestion
        if (Tools::getValue('token_cart') != Tools::hash('recover_cart_' . $id_cart)) {
```

Reply:
Why ?

Reply:
there are modules that use this feature to recover a cart, they're creating a link to `token_cart = hash`, and now you've changed the way it's being hashed

Reply:
Just no, nothing has changed , see above

Reply:
oh, yes, you're right, for some reason I thought that `hash` is doing something other than `md5` 🙄

Reply:
I think it's not worth blocking

Reply:
We can do it globally in another PR ;-)